src/index.ts

```diff
@@ -433,12 +466,12 @@ export class Helper {
    * @returns {undefined} void
    * @memberof CouchbaseHelper
    */
-  public async set(topic: string, index: archivist.IArchivistIndex, data: any, options?: couchbase.UpsertOptions) {
+  public async set<ICouchbaseStateObject>(topic: string, index: archivist.IArchivistIndex, data: any, options?: couchbase.UpsertOptions) {
```

Comment:
I only just noticed this, but if async/await in TypeScript works just like in JS, when you return a Promise you shouldn't mark your function `async`.